recipes/libjxl/all/conanfile.py

```diff
@@ -13,90 +15,115 @@ class LibjxlConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/libjxl/libjxl"
     topics = ("image", "jpeg-xl", "jxl", "jpeg")
-
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
-
-    exports_sources = "CMakeLists.txt", "patches/**"
-    generators = "cmake"
-    _cmake = None
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _minimum_cpp_standard(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+            # prevent hidden symbols of highway which are referenced by DSO
+            self.options["highway"].shared = True
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("brotli/1.0.9")
-        self.requires("highway/0.12.2")
-        self.requires("lcms/2.11")
+        if Version(self.version) < "0.7.0":
+            self.requires("highway/0.12.2")
+        else:
+            self.requires("highway/1.0.2")
+        self.requires("lcms/2.14")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        highway = self.dependencies["highway"]
+        if not highway.options.shared and self.options.shared:
```

Comment:
```suggestion
        if hasattr(highway, "options.shared") and not highway.options.shared and self.options.shared:
```
Reading highway recipe, older versions has no shared option.